File: 136-single-number/136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # naive solution
        # for i in nums:
        #     if nums.count(i) == 1:
        #         return i
        
        # better solution: XOR operator
        result = 0
        for num in nums:
            result = result ^ num
        return result

File: 136-single-number/136-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
// not linear runtime
// var singleNumber = function(nums) {
//     nums.sort((a,b) => a - b);
//     for (let i = 0; i < nums.length; i++){
//         if (nums[i] !== nums[i+1]) {
//             return nums[i];
//         } else {
//             i++;
//         }
//     }
// };

var singleNumber = function(nums) {
    let result = 0;
    for (let i = 0; i < nums.length; i++){
        result = result ^ nums[i];
    }
    return result;
};